addons/godot-xr-tools/objects/collision_hand/collision_hand.gd

```diff
@@ -0,0 +1,231 @@
+@tool
+## adds collision to the Open_XR Hands
+## ________________________________________________________
+## a CharacterBody3D that follows the corresponding XRController3D.
+## it is able to collide with the world and with static objects
+## ________________________________________________________
+## OPTIONAL FEATURES:
+## ________________________________________________________
+## collision on pickables/ weight on pickables/ zero_g movement
+
+class_name XRToolsCollisionHand
+extends CharacterBody3D
+
+#
+# THIS addition adds a KinematicBody that "chases" the controller THIS is assigned
+# to. THIS has a simple CollisionShape to simulate the empty hand and will copy
+# the CollisionShapes of held objects and add them to THIS as children; then it
+# will delete the copied CollisionShapes on drop.
+
+@export_group("Collision Hand Setup")
+
+## select controller
+@export var controller_path : NodePath
+
+## select visual hand
+@export var _hand_path : NodePath
+
+## select function_pickup
+@export var _pickup_path : NodePath #Should be child of ARVRController
```

Comment:
I would rework these to get the nodes themselves, not the path to the node. It allows for class limiting of the node so that the user can't put in an incorrect node (It also allows for highlighting when selecting a node directly.)